ui/src/pages/AllStories.tsx

```diff
@@ -1,18 +1,82 @@
 import AppBar from "@material-ui/core/AppBar";
 import Checkbox from "@material-ui/core/Checkbox";
+import { makeStyles } from "@material-ui/core/styles";
 import Switch from "@material-ui/core/Switch";
 import Tab from "@material-ui/core/Tab";
 import Tabs from "@material-ui/core/Tabs";
-import { useState } from "react";
+import AddIcon from "@material-ui/icons/Add";
+import RemoveIcon from "@material-ui/icons/Remove";
+import { useEffect, useReducer, useState } from "react";
 import * as React from "react";
 import styled from "styled-components";
-import useSWR from "swr";
+import useSWR, { mutate } from "swr";
 
 import { StoryDrawer } from "../components";
 import { a11yProps, AllStoriesTabs } from "../components/AllStoriesTabs";
 import VirtualizedTable from "../components/VirtualizedTable";
+import { colors } from "../styles/colors";
+import { fontSize, StyledAllStoriesHeader } from "../styles/typography";
 import { Story } from "../types/index";
 
+const StyledSwitch = styled(Switch)`
+  && {
+    .MuiSwitch-colorPrimary {
+      color: ${colors.white};
+    }
+    .MuiSwitch-track {
+      background-color: ${colors.primaryDark1};
+    }
+  }
+`;
+
+const StyledEmptyMessage = styled.div`
+  margin-left: 50vw;
+  margin-top: 30vh;
+  font-family: Poppins;
+  font-size: ${fontSize.subtitle};
+`;
+const StyledContainer = styled.div`
+  background-color: ${colors.primaryLight6};
+`;
+
+const useStyles = makeStyles({
+  root: {
+    background: colors.primaryLight6,
+    fontFamily: "Poppins",
+    fontStyle: "normal",
+    fontWeight: 500,
+    fontSize: "16px",
+    lineHeight: "150%",
+    color: colors.primaryDark1,
+    paddingLeft: "64px",
+    boxShadow: "none",
+  },
+  indicator: {
+    backgroundColor: colors.primaryDark1,
+  },
+  checkbox: {
+    "&$checked": {
+      color: colors.primaryDark1,
+    },
+  },
+  checked: {},
+});
+
+export interface StoryView {
```

Comment:
Can we make `StoryView` a union type? so something like:

```ts
type StoryView = Story & {
  // whatever fields have changed or been added
}
```

Reply:
Isn't intersection types for adding together different pre-existing types? in that case `StoryView` would end up having everything on `Story` +  the member `author_name` which we don't want since this would mean returning extra longitude, latitude, etc in `createData`?

Reply:
Ah true. Can we make use of the `Omit` or `Pick` type then? https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys